lib/logstash/outputs/elasticsearch/common_configs.rb

```diff
@@ -75,13 +75,8 @@ def self.included(mod)
       # to prevent LS from sending bulk requests to the master nodes.  So this parameter should only reference either data or client nodes in Elasticsearch.
       mod.config :hosts, :validate => :array, :default => ["127.0.0.1"]
 
-      # This plugin uses the bulk index API for improved indexing performance.
-      # This setting defines the maximum sized bulk request Logstash will make
-      # You you may want to increase this to be in line with your pipeline's batch size.
-      # If you specify a number larger than the batch size of your pipeline it will have no effect,
-      # save for the case where a filter increases the size of an inflight batch by outputting
-      # events.
-      mod.config :flush_size, :validate => :number, :default => 500
+      # DEPRECATED. We now restrict bulk sizes to a size that balances heap size with reducing round-trips. Very large events will be sent in a single request by themselves.
+      mod.config :flush_size, :validate => :number, :default => 500, :deprecated => "No Longer Needed"
```

Comment:
You make a good point, but I think this is still a good use of deprecate. In this case we don't want to break configs, but we want people to stop explicitly setting this. We could leave this feature in, but that's bad because it does things no one should want to do.

I'm open to hearing a conflicting opinion here of course, but there's no good reason to set `flush_size` once this PR lands IMHO.